engine/verification/test/helper.go

```diff
@@ -468,7 +468,8 @@ func VerifiableDataChunk(chunkIndex uint64, er utils.CompleteExecutionResult) *v
 	var endState flow.StateCommitment
 	// last chunk
 	if int(chunkIndex) == len(er.Receipt.ExecutionResult.Chunks)-1 {
-		endState = er.Receipt.ExecutionResult.FinalStateCommit
+		finalState, _ := er.Receipt.ExecutionResult.FinalStateCommitment()
+		endState = finalState
```

Comment:
`VerifiableDataChunk` is a test helper that previously would not deal with any unhappy path. However, after this modification, there is an expected unhappy path case, which we should capture. Otherwise, upon using this function in tests, it can fail silently without indicating the cause. I would suggest the following refactoring:
1. Adding a `t *testing.T` as an input parameter: 
`VerifiableDataChunk(t *testing.T, chunkIndex uint64, er utils.CompleteExecutionResult) *verification.VerifiableChunkData`

2. Fail the test on a `require` if the final state commitment is not retrievable: 
```
finalState, ok := er.Receipt.ExecutionResult.FinalStateCommitment()
require.True(t, ok)
```

Reply:
Thanks Yahya, implemented!